contracts/src/ZKEmailProof.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import {ERC721URIStorage} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import {Base64} from "@openzeppelin/contracts/utils/Base64.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+
+struct ZKEmailProofMetadata {
+    uint256 blueprintId;
+    uint256[] proof;
+    address verifier;
+    string publicOutputs;
```

Comment:
Wasn't sure on correct format here and ran out of time. Perhaps this would be an array of key value strings or similar and then we could add them to the metadata in the same way that `proofJson` is added in `tokenURI()`

Reply:
proof is going to be [uint[2], uint[2][2], uint[2]] always, publicOutputs is going to be uint256[], then we also need a decoded public output field which would be included in the token uri

Reply:
**TODO notes**
1. Use `[uint[2], uint[2][2], uint[2]]` for proof
2. Loop over public (`uint256[]`) outputs the same way the proof currently works
3. Add decoded public outputs which should be flattened json